core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -465,9 +465,7 @@ private[kafka] class Processor(val id: Int,
         // letting a processor exit might cause a bigger impact on the broker. Usually the exceptions thrown would
         // be either associated with a specific socket channel or a bad request. We just ignore the bad socket channel
         // or request. This behavior might need to be reviewed if we see an exception that need the entire broker to stop.
-        case e: ControlThrowable => throw e
```

Comment:
Maybe worth revising this comment as it seems like we should never hit this exception since each method catches exceptions itself.

Reply:
Updated comment.